### PR TITLE
Merchant index stats#22

### LINKED ISSUE #22: User Story 46, Merchant Index Page Statistics
```
As a visitor
When I visit the merchants index page, I see an area with statistics:
- top 3 merchants who have sold the most by price and quantity
- top 3 merchants who were fastest at fulfilling items in an order
- worst 3 merchants who were slowest at fulfilling items in an order
- top 3 states where any orders were shipped
- top 3 cities where any orders were shipped (Springfield, MI should not be grouped with Springfield, CO)
- top 3 biggest orders by quantity of items
```

### Checklist:

- [ ] base code tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #24

depends on #23

### PR DESCRIPTION
**Pull request by asmolentzov**
Completes User Story 46: Merchant Index Page Statistics

- Adds a Statistics area to the merchants index page with the following content
- Top three merchants who sold the most by price
  - This takes into account that the merchant is enabled and the order_item is fulfilled
- Top three merchants who sold the most by quantity
  - This takes into account that the merchant is enabled and the order_item is fulfilled
- Fastest 3 merchants at fulfilling items
- Slowest 3 merchants at fulfilling items
  - Fastest and Slowest are both taken from the same method: `merchants_by_time` that returns a sorted array of merchants by the time it takes them to fulfill an item. I add logic in the controller to limit it to three (and reverse it for slowest), but can also just push that logic to the model if that would be better. 
  - Takes into account that the merchant is enabled and that the order_items are fulfilled
- Top 3 states where any orders were shipped
  - Takes into account that the order status is 1, fulfilled.
  - Does use a Ruby method (`.map`) to get the specific info we want (the states). Not sure if there is a better way to do this. 
- Top 3 cities where any orders were shipped (taking state into account so e.g. all 'Springfields' in all states are not lumped together)
  - Takes into account that the order status is 1, fulfilled.
  -Does use a Ruby method (`.map`) to get the specific info we want (the cities and states). Not sure if there is a better way to do this. 
- Top 3 biggest orders by quantity of items
  - This is the only one that is not in the User model - this is in the Order model. 
  - Takes into account that the status is 1 - fulfilled.
  - Takes into account that the order_items are fulfilled
- All tests passing

Closes #22

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -7,6 +7,13 @@ def index
     else
       @merchants = User.enabled_merchants
     end
+    @top_merchants_by_quantity = User.merchants_by_quantity
```
Comment (danbriechle) [inline app/controllers/users_controller.rb]: Not that its needed but this might be a good place for a refactor.

File: spec/features/merchant_index_page_spec.rb [hunk 2]
```diff
@@ -27,5 +27,189 @@
...
+      
+      visit merchants_path
+      
+      within "#statistics" do
+        expect(page).to have_content("Biggest Orders by Quantity of Items:#{top_orders}")
+        expect(page).to_not have_content("Order ##{order_4.id}")
+        expect(page).to_not have_content("Order ##{order_5.id}")
+        expect(page).to_not have_content("Order ##{order_6.id}")
+      end
+    end
```
Comment (danbriechle) [inline spec/features/merchant_index_page_spec.rb]: This looks really solid the only sad path I can think of is create an admin an unregistered user and see if they throw anything. Really nice work !

File: spec/models/order_item_spec.rb [hunk 3]
```diff
@@ -11,6 +11,6 @@
     it { should validate_presence_of(:item_id)}
     it { should validate_presence_of(:quantity)}
     it { should validate_presence_of(:order_price)}
-    it { should validate_presence_of(:fulfilled)}
+    it { should validate_inclusion_of(:fulfilled).in_array([true, false])}
```
Comment (danbriechle) [inline spec/models/order_item_spec.rb]: 👍

**Comment by danbriechle** [comment]
Great work ! Happy to merge when everybody else gets a good look ! Way to crush it !

File: spec/features/merchant_index_page_spec.rb (hunk 2 above)
Comment (Bradniedt) [reply spec/features/merchant_index_page_spec.rb]: I'll echo what Dan says here - tough to think of expectations for robustness as these are well-written. Splitting hairs would be testing that maybe non-enabled merchants don't show up? Solid testing!!

**Comment by Bradniedt** [comment]
Phew - awesome work, Anna! I love the addition of fairbanks - (alaska is on my bucket list) - and your tests are well-written. The logic in the AR queries is sound, and I think using ruby to make our output work is fine because really we have all the data we need there, and the ruby just helps us widdle it down. Great work!!

**Comment by Bradniedt** [comment]
Forgot to add - I approve the merge!

File: app/views/users/index.html.erb [hunk 4]
```diff
@@ -1,3 +1,56 @@
+<div id="statistics">
+  Statistics!
+  <p>Top Merchants by Quantity: 
+    <% @top_merchants_by_quantity.each do |merchant| %>
```
Comment (saleratus) [inline app/views/users/index.html.erb]: Very minor: is "Statistics!" (with no tags) supposed to be here, or is it an artifact of earlier work?

Comment (asmolentzov) [reply app/views/users/index.html.erb]: It's just a non-tagged header for the display - can be edited/removed later!
